sdk/python/feast/inference.py

```diff
@@ -45,12 +48,72 @@ def infer_entity_value_type_from_feature_views(
                     entity.value_type != ValueType.UNKNOWN
                     and entity.value_type != inferred_value_type
                 ) or (len(extracted_entity_name_type_pairs) > 1):
-                    raise ValueError(
+                    raise RegistryInferenceFailure(
+                        "Entity",
                         f"""Entity value_type inference failed for {entity_name} entity.
-                        Multiple viable matches. Please explicitly specify the entity value_type
-                        for this entity."""
+                        Multiple viable matches.
+                        """,
                     )
 
                 entity.value_type = inferred_value_type
 
     return entities
+
+
+def infer_event_timestamp_column_for_data_sources(
+    data_sources: List[DataSource],
+) -> List[DataSource]:
```

Comment:
Nit, is there any reason for this to return the same datasources? Since the method just updates them in-place. Also rename to `update_data_sources_with_event_timestamp_column` or something?

Reply:
sounds good